samples/cellular/modem_shell/src/sms/sms_shell.c

```diff
@@ -161,34 +136,85 @@ static int sms_shell(const struct shell *shell, size_t argc, char **argv)
 		}
 	}
 
-	/* Run given command with it's arguments */
-	switch (command) {
-	case SMS_CMD_REGISTER:
-		err = sms_register();
-		break;
-	case SMS_CMD_UNREGISTER:
-		err = sms_unregister();
-		break;
-	case SMS_CMD_SEND:
-		err = sms_send_msg(arg_number, arg_message, arg_message_type);
-		break;
-	case SMS_CMD_RECV:
-		err = sms_recv(arg_receive_start);
-		break;
-	default:
-		mosh_error("Internal error. Unknown sms command=%d", command);
-		err = -EINVAL;
-		break;
+	err = sms_send_msg(arg_number, arg_message, arg_message_type);
```

Comment:
`sms send` without options prints "Number not given". Wouldn't it be better to just print the help instead? Now user has to type "sms send -h/--help" to see the command syntax.

Reply:
`ping` and `rest` will print failure and help. In linux world, I guess an error print with the plain command is more common although many commands has functionality without params but ping in Unix and Windows is an example.

We tend to print the help in many places when there are no arguments probably mainly because that's the concept with Zephyr shell macros.

So I guess the question is do we print error and help or just help?

Reply:
My point is that `sms send` prints ONLY the error, not the help. `ping` and `rest` commands without parameters print an error, but also the help. I'm fine with printing the error and the help with `sms send`.

Reply:
It prints now just the help.

Reply:
I've now actually change `ping` and `rest` commands also not to print errors but just usage if no options are given.